code/game/objects/items/weapons/dnascrambler.dm

```diff
@@ -55,3 +55,36 @@
 	used = 1
 	update_icon()
 	name = "used " + name
+
+/obj/item/dnascrambler/renamer
+	name = "dna renamer"
+	desc = "An illegal genetic serum designed to alter the user's identity."
+
+/obj/item/dnascrambler/renamer/attack(mob/M as mob, mob/user as mob)
+	if(!M || !user)
+		return
+
+	if(!ishuman(M) || !ishuman(user))
+		return
+
+	if(used)
+		return
+
+	if(M == user)
+		user.visible_message("<span class='danger'>[user] injects [user.p_them()]self with [src]!</span>")
+		injected(user, user)
+
+/obj/item/dnascrambler/renamer/injected(var/mob/living/carbon/human/target, var/mob/living/carbon/user)
+	var/newname = sanitize(copytext(input(target, "What would you like your new identity to be?", "Name change") as null|text,1,MAX_NAME_LEN))
```

Comment:
You probably need something above this (or do `if(used) return` below this), otherwise you can click yourself multiple times to get multiple uses out of this